### PR TITLE
7412: Populate checkboxes

### PR DESCRIPTION
**Pull request by kat1906**
Story: [AB#7412](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7412)
Task: [AB#7566](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7566)

### CODE REVIEW

File: app/pages/sections/service-recipients/template.test.js [hunk 1]
```diff
@@ -47,21 +66,61 @@ describe('service-recipients page', () => {
...
+      harness.request(context, ($) => {
+        const checkbox1 = $('[data-test-id="organisation-name-checkbox-ods1"]');
+        expect(checkbox1.length).toEqual(1);
+        expect(checkbox1.text().trim()).toEqual(context.tableData[0].organisationName.text);
+
+        const checkboxInput1 = checkbox1.find('input');
+        expect(checkboxInput1.attr('id')).toEqual(context.tableData[0].organisationName.id);
+        expect(checkboxInput1.attr('name')).toEqual(context.tableData[0].organisationName.name);
+        expect(checkboxInput1.attr('type')).toEqual('checkbox');
+        expect(checkboxInput1.attr('checked')).toEqual(undefined);
```
Comment (kat1906) [inline app/pages/sections/service-recipients/template.test.js:88]: checked is `undefined` if not there and has a value of `''` (empty string) if it is there

File: app/pages/sections/service-recipients/ui-tests/withSavedData.ui.test.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    .expect(await extractInnerText(checkbox1Label)).eql(mockOapiData[0].name)
+    .expect(checkbox1Label.getAttribute('for')).eql('ods1-id')
+    .expect(checkbox2Input.exists).ok()
+    .expect(checkbox2Input.getAttribute('name')).eql('ods2-name')
+    .expect(checkbox2Input.getAttribute('id')).eql('ods2-id')
+    .expect(checkbox2Input.getAttribute('type')).eql('checkbox')
+    .expect(checkbox2Input.getAttribute('checked')).eql('')
+    .expect(checkbox2Label.exists).ok()
+    .expect(await extractInnerText(checkbox2Label)).eql(mockOapiData[1].name)
+    .expect(checkbox2Label.getAttribute('for')).eql('ods2-id');
```
Comment (alibombaye) [inline app/pages/sections/service-recipients/ui-tests/withSavedData.ui.test.js:86]: can you blank lines in between checkbox 1 and 2 please

File: app/pages/sections/service-recipients/ui-tests/withSavedData.ui.test.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  const checkbox1Input = Selector('[data-test-id="organisation-name-checkbox-ods1"] input');
+  const checkbox1Label = Selector('[data-test-id="organisation-name-checkbox-ods1"] label');
+  const checkbox2Input = Selector('[data-test-id="organisation-name-checkbox-ods2"] input');
+  const checkbox2Label = Selector('[data-test-id="organisation-name-checkbox-ods2"] label');
+  await t
+    .expect(checkbox1Input.exists).ok()
+    .expect(checkbox1Input.getAttribute('name')).eql('ods1-name')
+    .expect(checkbox1Input.getAttribute('id')).eql('ods1-id')
+    .expect(checkbox1Input.getAttribute('type')).eql('checkbox')
+    .expect(checkbox1Input.getAttribute('checked')).eql('')
```
[the remark this answers is not in the document]
Comment (alibombaye) [reply app/pages/sections/service-recipients/ui-tests/withSavedData.ui.test.js:75]: would something like this not work?
`.find('input:checked').exists).ok();`

**Comment by alibombaye** [comment]
much prefer the ui test break down like this 👍

**Pull request merged**